metagraph/plugins/numpy/types.py

```diff
@@ -120,6 +120,18 @@ def __init__(self, data, mask=None):
             if mask.shape != data.shape:
                 raise ValueError("mask must be the same shape as data")
 
+    @property
+    def shape(self):
+        return self.value.shape
+
+    def as_dense(self, fill_value=0, copy=False) -> np.ndarray:
+        vector = self.value
+        if copy:
```

Comment:
This should be `if copy or self.mask is not None`. Even if the user didn't ask for a copy, we never want to mutate the original object.

Reply:
The same applies down to NumpyMatrix.as_dense as well

Reply:
https://github.com/metagraph-dev/metagraph/pull/136/commits/9541c9057a969101646a6ff571242c4696bd627d implements this suggestion.